pumps/elasticsearch.go

```diff
@@ -90,57 +143,85 @@ func (e *ElasticsearchPump) WriteData(data []interface{}) error {
 		"prefix": elasticsearchPrefix,
 	}).Info("Writing ", len(data), " records")
 
-	if e.esClient == nil {
+	if e.operator == nil {
 		log.WithFields(logrus.Fields{
 			"prefix": elasticsearchPrefix,
 		}).Debug("Connecting to analytics store")
 		e.connect()
 		e.WriteData(data)
 	} else {
 		if len(data) > 0 {
+			e.operator.processData(data, e.esConf)
+		}
+	}
+	return nil
+}
+
+func GetIndexName(esConf *ElasticsearchConf) string {
+	var indexName = esConf.IndexName
+
+	if esConf.RollingIndex {
+		var currentTime = time.Now()
+		//This formats the date to be YYYY.MM.DD but Golang makes you use a specific date for its date formatting
+		indexName += "-" + currentTime.Format("2006.01.02")
+	}
+	return indexName
+}
+
+func GetMapping(datum interface{}, extendedStatistics bool) map[string]interface{} {
+	var record, _ = datum.(analytics.AnalyticsRecord)
```

Comment:
You sure you want to do the `, _` version? This will not panic if the type is not what you expect.

And if you always want the type to be exactly this, either make it the parameter type or error if the type is incorrect.